homeassistant/components/epson/media_player.py

```diff
@@ -59,17 +59,17 @@
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_PORT, default=80): cv.port,
     }
 )
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Epson projector from a config entry."""
-    unique_id = config_entry.entry_id
-    projector = hass.data[DOMAIN][unique_id]
+    entry_id = config_entry.entry_id
+    unique_id = config_entry.unique_id
+    projector = hass.data[DOMAIN][entry_id]
     projector_entity = EpsonProjectorMediaPlayer(
-        projector, config_entry.title, unique_id
+        projector, config_entry.title, unique_id if unique_id else entry_id
```

Comment:
It's not a good solution to mix different kinds of unique ids. Then we're removing the point of the unique_id. It will no longer be able to match the same device to the entity and resolve different devices.

Can we try to get the serial number when we have connected to the device and update the unique_id of the config entry and then reload the integration?

Reply:
I think it's the best solution, but I don't know how to reload integration if device is `on` and `unique_id` is not set. any examples?

Reply:
Use `hass.config_entries.async_update_entry` to update the config entry data and `hass.config_entries.async_reload` to reload the integration.

Reply:
can I do something like this in async_update in `media_player`:
```
if power_state == EPSON_CODES[POWER]:
  self._state = STATE_ON
  if not self._unique_id and self._entry:
      unique_id = await self._projector.get_serial()
      self.hass.config_entries.async_update_entry(
          self._entry, unique_id=unique_id
      )
      self.hass.config_entries.async_reload(self._entry.entry_id)
```

Reply:
Yes. `self._entry` should always be there I think. Possibly create a task that reloads the integration instead of awaiting it.

Reply:
Can you check if approach in current commit is something you had in mind?